packages/dds/tree/src/feature-libraries/optional-field/optionalFieldChangeTypes.ts

```diff
@@ -23,7 +23,7 @@ export type RegisterId = ChangeAtomId | "self";
  * Each register is identified using a {@link RegisterId}.
  * The active register holds the current value of the field, and other registers hold detached roots.
  */
-export interface OptionalChangeset {
+export interface OptionalChangeset<TChildChange = NodeChangeset> {
```

Comment:
Is this type parameter truly unconstrained?

Reply:
It can't be undefined but that's not something we typically try to enforce through the type system.

Reply:
Right, I guess I'm asking why this type is so open. How many change types do we expect to be here in production?

Reply:
(feel free to merge and not let this block)

Reply:
In production, only NodeChangeset is used.
The open type is so that we can use something else (e.g. TestChange) than NodeChangeset when we're testing a field kind implementation. This allows the test code for a field not to have to depend on the detail of NodeChangeset. Also TestChange comes with some helpful properties that make it better for detecting issues in whatever is using it.

Ya, I would have merged already if it weren't for prettier messing with the snapshot files. Should be resolved now.